### ISSUE TITLE
Beta 6 error on SPTAudioStreamingController construction using swift

### ISSUE THREAD

**Issue by rahmen86** [spotify/ios-sdk#195]
After upgrading the iOS SDK from Beta 5 to Beta 6 I receive this error:

Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[SPTAudioStreamingController initWithClientId:]: unrecognized selector sent to instance 0x7a3f15a9740'

The line that seems to give the error is this one:

self.player = SPTAudioStreamingController(clientId: "my client id")
In Beta 5 it was not necessary to provide the clientId. After the upgrade I added the clientId string and get the error above.

Note that the exact same code (without the clientId argument) worked with Beta 5. I get the same error no matter what string I provide. Is this a SDK related bug? SDK in Swift? Or do I have to do additional things after upgrading to Beta 6?

**Comment by possan** [comment]
I'm not that familiar with Swift, but don't you have to also call alloc somewhere there, like this in objective-c: `[[SPTAudioStreamingController alloc] initWithClientID:@"something"]`

**Comment by rahmen86** [comment]
I made it work now and I assume that there was an error in Beta 6 or maybe in the previous Xcode version. I still think that the errors are a bit misleading so I decided to write what happened for me.

After upgrading to iOS SDK Beta 7 and upgrading Xcode to version 6.2 I created a new project in swift. Writing `var a = SPTAudioStreamingController()` created an error saying "...use initWithClientId".

I assumed I should try this instead `var a = SPTAudioStreamingController.initWithClientId("client-id-string")` but this showed another error saying the method doesn't exist.

I was confused for a while since I forgot to add the client ID in the constructor. This is how it should be:
`var a = SPTAudioStreamingController(clientId: "client-id-string")`

I hope this helps someone else in the future.
